test.ts

```diff
@@ -0,0 +1,139 @@
+import { which } from "./deps/deno_which.ts"
+import { unimplemented } from "./deps/std/assert.ts"
+import { parse as parseFlags } from "./deps/std/flags.ts"
+import { blue, dim, gray, green, red, yellow } from "./deps/std/fmt/colors.ts"
+import { walk } from "./deps/std/fs.ts"
+import { Buffer, readLines } from "./deps/std/io.ts"
+import * as path from "./deps/std/path.ts"
+import { readerFromStreamReader, writeAll } from "./deps/std/streams.ts"
+import { parse as parseFrontmatter } from "./frontmatter.ts"
+
+const { _, reload, ...rest } = parseFlags(Deno.args, {
+  alias: {
+    c: "concurrency",
+    p: "project",
+    r: "reload",
+  },
+  string: ["concurrency", "project", "reload"],
+  default: {
+    concurrency: Infinity,
+  },
+})
+
+const include: string[] = []
+for await (
+  const { path: pathname } of walk(".", {
+    exts: [".ts", ".tsx"],
+    followSymlinks: true,
+    includeDirs: false,
+    match: _.map((value) => {
+      if (typeof value !== "string") {
+        throw new Error(
+          `Specified an invalid include \`${pathname}\` (expected a glob or path to example file)`,
+        )
+      }
+      return path.isGlob(value) ? path.globToRegExp(value) : new RegExp(value)
+    }),
+  })
+) include.push(pathname)
+
+const concurrency = +rest.concurrency
+
+const project = rest.project ?? await (async () => {
+  for (const pathname of ["deno.json", "deno.jsonc"]) {
+    try {
+      return JSON.parse(await Deno.readTextFile(pathname))
+    } catch (_e) {}
+  }
+  return
+})()
+const _importMapURL = project.importMap
+  ? path.toFileUrl(path.join(Deno.cwd(), project.importMap))
+  : undefined
+
+const browser = typeof rest.browser === "string"
+  ? rest.browser === "" ? await which("chromium") : rest.browser
+  : undefined
+if (browser === "") throw new Error(`Failed to detect chromium path. Specify via \`--browser\`.`)
+
+const controller = new AbortController()
+const { signal } = controller
+const queue: (() => Promise<void>)[] = []
+const failed: string[] = []
+let done = 0
+
+include.forEach((pathname) =>
+  queue.push(async () => {
+    const { frontmatter } = parseFrontmatter(pathname, await Deno.readTextFile(pathname), {
+      test_skip(value) {
+        return value !== undefined
+      },
+    })
+    const quotedPathname = `"${pathname}"`
+    if (frontmatter.test_skip) {
+      console.log(yellow("Skipping"), quotedPathname)
+      done++
+      return
+    }
+    console.log(gray("Testing"), quotedPathname)
+    const logs = new Buffer()
+    const code = await (browser ? runBrowser : runDeno)(pathname, logs)
+    done++
+    const progress = dim(`(${done}/${include.length})`)
+    if (code) {
+      failed.push(pathname)
+      console.log(red("Failed"), progress, quotedPathname)
+      console.groupCollapsed()
```

Comment:
do groups work in a normal terminal? I haven't seen these used outside of browser

Reply:
Not sure... thought maybe they'd be respected in GH workflow logs... but perhaps not.